--- a/src/command.coffee
+++ b/src/command.coffee
@@ -1,4 +1,5 @@
 child_process = require 'child_process'
+fs = require 'fs'
 path = require 'path'
 _ = require 'underscore-plus'
 semver = require 'npm/node_modules/semver'
@@ -90,6 +91,10 @@
         version = @normalizeVersion(version)
         @installedAtomVersion = version if semver.valid(version)
 
+      if electronVersion is undefined
+        try
+          electronVersion = fs.readFileSync('/usr/<lib>/electron/version', 'utf8').slice(1)
+
       # TODO Remove ATOM_NODE_VERSION env var support after a couple releases
       @electronVersion = process.env.ATOM_ELECTRON_VERSION ? process.env.ATOM_NODE_VERSION ? electronVersion ? '0.22.0'
 
