--- a/src/module-cache.coffee
+++ b/src/module-cache.coffee
@@ -194,14 +194,19 @@
   return
 
 registerBuiltins = (devMode) ->
-  electronRoot = path.join(process.resourcesPath, 'atom.asar')
+  fs = require 'fs-plus'
+  electronRoot = path.join(process.resourcesPath, 'electron.asar')
 
   commonRoot = path.join(electronRoot, 'common', 'api', 'lib')
+  if not fs.existsSync(commonRoot)
+    commonRoot = path.join(electronRoot, 'common', 'api')
   commonBuiltins = ['callbacks-registry', 'clipboard', 'crash-reporter', 'screen', 'shell']
   for builtin in commonBuiltins
     cache.builtins[builtin] = path.join(commonRoot, "#{builtin}.js")
 
   rendererRoot = path.join(electronRoot, 'renderer', 'api', 'lib')
+  if not fs.existsSync(rendererRoot)
+    rendererRoot = path.join(electronRoot, 'renderer', 'api')
   rendererBuiltins = ['ipc', 'remote']
   for builtin in rendererBuiltins
     cache.builtins[builtin] = path.join(rendererRoot, "#{builtin}.js")
