diff --git a/bootstrap.py b/bootstrap.py
index e526ff2..c887015 100755
--- a/bootstrap.py
+++ b/bootstrap.py
@@ -5,10 +5,10 @@
 from __future__ import print_function
 import os, sys, stat, hashlib, subprocess
 
-WAFRELEASE = "waf-1.8.12"
+WAFRELEASE = "waf-1.7.16"
 WAFURLS    = ["https://waf.io/" + WAFRELEASE,
               "http://www.freehackers.org/~tnagy/release/" + WAFRELEASE]
-SHA256HASH = "01bf2beab2106d1558800c8709bc2c8e496d3da4a2ca343fe091f22fca60c98b"
+SHA256HASH = "b64dc26c882572415fd450b745006107965f3fe17b357e3eb43d6676c9635a61"
 
 if os.path.exists("waf"):
     wafver = subprocess.check_output([sys.executable, './waf', '--version']).decode()
@@ -44,8 +44,6 @@ if SHA256HASH == hashlib.sha256(waf).hexdigest():
     print("Checksum verified.")
 else:
     print("The checksum of the downloaded file does not match!")
-    print(" - got:      {}".format(hashlib.sha256(waf).hexdigest()))
-    print(" - expected: {}".format(SHA256HASH))
     print("Please download and verify the file manually.")
 
     sys.exit(1)
diff --git a/waftools/generators/headers.py b/waftools/generators/headers.py
index 3c83292..7401f01 100644
--- a/waftools/generators/headers.py
+++ b/waftools/generators/headers.py
@@ -1,5 +1,5 @@
 def __cp_to_variant__(ctx, variant, basename):
-    src = ctx.bldnode.search_node(basename).read()
+    src = ctx.bldnode.search(basename).read()
     node = ctx.bldnode.make_node("{0}/{1}".format(variant, basename))
     node.parent.mkdir()
     node.write(src)
diff --git a/waftools/waf_customizations.py b/waftools/waf_customizations.py
index fb5f130..5a9bf27 100644
--- a/waftools/waf_customizations.py
+++ b/waftools/waf_customizations.py
@@ -40,6 +40,15 @@ def build(ctx):
                 run_str = cls.hcode + '${LAST_LINKFLAGS}'
             except TypeError:
                 run_str = cls.hcode.decode('iso8859-1') + '${LAST_LINKFLAGS}'
+        def __str__(self):
+            tgt_str = ' '.join([a.nice_path() for a in self.outputs])
+            return 'linking -> {0}\n'.format(tgt_str)
+
+    cls = Task.classes['cshlib']
+    class cshlib(cls):
+        def __str__(self):
+            tgt_str = ' '.join([a.nice_path() for a in self.outputs])
+            return 'linking -> {0}\n'.format(tgt_str)
 
     cls = Task.classes['macplist']
     class macplist(cls):
diff --git a/wscript b/wscript
index d96f74b..9d0391a 100644
--- a/wscript
+++ b/wscript
@@ -897,7 +897,7 @@ def is_debug_build(ctx):
 
 def configure(ctx):
     ctx.resetenv(ctx.options.variant)
-    ctx.check_waf_version(mini='1.8.4')
+    ctx.check_waf_version(mini='1.7.15')
     target = os.environ.get('TARGET')
     (cc, pkg_config, ar, windres) = ('cc', 'pkg-config', 'ar', 'windres')
 
