--- a/source/CMakeLists.txt
+++ b/source/CMakeLists.txt
@@ -165,7 +169,7 @@
     add_definitions(-D__STDC_LIMIT_MACROS=1)
     add_definitions(-std=gnu++98)
     if(ENABLE_PIC)
-         add_definitions(-fPIC)
+         add_definitions(-fPIC -DPIC)
     endif(ENABLE_PIC)
     if(NATIVE_BUILD)
         if(INTEL_CXX)
--- a/source/test/checkasm-a.asm
+++ b/source/test/checkasm-a.asm
@@ -156,7 +156,7 @@
     mov  r9, rax
     mov r10, rdx
     lea  r0, [error_message]
-    call puts
+    call puts wrt ..plt
     mov  r1, [rsp+max_args*8]
     mov  dword [r1], 0
     mov  rdx, r10
