From 8ed454a22637a21d29d2300874ac84b0d7ba4152 Mon Sep 17 00:00:00 2001
From: Hualet Wang <mr.asianwang@gmail.com>
Date: Thu, 19 Jan 2017 17:27:57 +0800
Subject: [PATCH] Add option to disable udpate module

Change-Id: Ib40bb92dc38a12f03f1c06f9a7f63a31e8f40b61
---

diff --git a/frame/frame.pro b/frame/frame.pro
index 41695b9..9f803ec 100644
--- a/frame/frame.pro
+++ b/frame/frame.pro
@@ -57,3 +57,7 @@
 
 RESOURCES += \
     frame.qrc
+
+isEqual(DISABLE_SYS_UPDATE, YES){
+    DEFINES += DISABLE_SYS_UPDATE
+}
diff --git a/frame/settingswidget.cpp b/frame/settingswidget.cpp
index ac75509..1a16c41 100644
--- a/frame/settingswidget.cpp
+++ b/frame/settingswidget.cpp
@@ -68,7 +68,9 @@
     loadModule(new mouse::MouseModule(this));
     loadModule(new KeyboardModule(this));
     loadModule(new wacom::WacomModule(this));
+#ifndef DISABLE_SYS_UPDATE
     loadModule(new UpdateModule(this));
+#endif
     loadModule(new SystemInfoModule(this));
 
     m_settingsWidget->setLayout(m_settingsLayout);
