diff -up wrk/configure.ac.wrk wrk/configure.ac
--- wrk/configure.ac.wrk	2013-07-12 17:19:34.079965200 +0200
+++ wrk/configure.ac	2013-07-22 16:53:01.630128139 +0200
@@ -414,8 +414,6 @@ AC_CHECK_HEADERS(lpc10.h, ,
     AC_CHECK_LIB(lpc10, create_lpc10_encoder_state, LPC10_LIBS="$LPC10_LIBS -llpc10", found_liblpc10=no)
 if test "$found_liblpc10" = yes; then
     AC_DEFINE(EXTERNAL_LPC10, 1, [Define if you are using an external LPC10 library])
-else
-    LIBLPC10_LIBADD=../lpc10/liblpc10.la
 fi
 AM_CONDITIONAL(EXTERNAL_LPC10, test x$found_liblpc10 = xyes)
 AC_SUBST(LIBLPC10_LIBADD)
@@ -603,7 +601,7 @@ AC_SUBST(PLAYRECLINKS)
 AM_CONDITIONAL(STATIC_LIBSOX_ONLY, test "$enable_shared" = "no" -a "$enable_static" = "yes")
 
 dnl Generate output files.
-AC_CONFIG_FILES(Makefile src/Makefile libgsm/Makefile lpc10/Makefile msvc9/Makefile msvc10/Makefile sox.pc)
+AC_CONFIG_FILES(Makefile src/Makefile libgsm/Makefile msvc9/Makefile msvc10/Makefile sox.pc)
 AC_OUTPUT
 
 if test "$using_gsm" != "no"; then
diff -up wrk/Makefile.am.wrk wrk/Makefile.am
--- wrk/Makefile.am.wrk	2013-07-12 17:17:11.512971143 +0200
+++ wrk/Makefile.am	2013-07-12 17:17:29.990970373 +0200
@@ -2,8 +2,8 @@
 
 ACLOCAL_AMFLAGS = -I m4
 
-SUBDIRS = lpc10 libgsm src
-DIST_SUBDIRS = lpc10 libgsm src msvc9 msvc10
+SUBDIRS = libgsm src
+DIST_SUBDIRS = libgsm src msvc9 msvc10
 
 RM = rm -f
 
