Index: OpenCC-ver.1.0.2/doc/CMakeLists.txt
===================================================================
--- OpenCC-ver.1.0.2.orig/doc/CMakeLists.txt
+++ OpenCC-ver.1.0.2/doc/CMakeLists.txt
@@ -29,7 +29,7 @@ if(BUILD_DOCUMENTATION)
     	DIRECTORY
     		${CMAKE_BINARY_DIR}/doc/html
     	DESTINATION
-    		${DIR_SHARE_OPENCC}doc
+    		${DIR_SHARE_OPENCC}/doc
     )
 
     set_directory_properties(
Index: OpenCC-ver.1.0.2/CMakeLists.txt
===================================================================
--- OpenCC-ver.1.0.2.orig/CMakeLists.txt
+++ OpenCC-ver.1.0.2/CMakeLists.txt
@@ -65,11 +65,11 @@
 ######## Directory
 
 set (DIR_PREFIX ${CMAKE_INSTALL_PREFIX})
-set (DIR_INCLUDE ${DIR_PREFIX}/include/)
-set (DIR_SHARE ${DIR_PREFIX}/share/)
-set (DIR_ETC ${DIR_PREFIX}/etc/)
+set (DIR_INCLUDE ${INCLUDE_INSTALL_DIR})
+set (DIR_SHARE ${SHARE_INSTALL_PREFIX})
+set (DIR_ETC ${SYSCONF_INSTALL_DIR})
 set (LIB_SUFFIX "")
-set (DIR_LIBRARY ${DIR_PREFIX}/lib${LIB_SUFFIX}/)
+set (DIR_LIBRARY ${LIB_INSTALL_DIR})

 if (DEFINED SHARE_INSTALL_PREFIX)
   set (DIR_SHARE ${SHARE_INSTALL_PREFIX})
@@ -82,8 +82,8 @@ if (DEFINED SYSCONF_INSTALL_DIR)
   set (DIR_ETC ${SYSCONF_INSTALL_DIR})
 endif (DEFINED SYSCONF_INSTALL_DIR)
 
-set (DIR_SHARE_OPENCC ${DIR_SHARE}opencc/)
-set (DIR_SHARE_LOCALE ${DIR_SHARE}locale/)
+set (DIR_SHARE_OPENCC ${DIR_SHARE}/opencc/)
+set (DIR_SHARE_LOCALE ${DIR_SHARE}/locale/)
 
 ######## Configuration
 
Index: OpenCC-ver.1.0.2/src/CMakeLists.txt
===================================================================
--- OpenCC-ver.1.0.2.orig/src/CMakeLists.txt
+++ OpenCC-ver.1.0.2/src/CMakeLists.txt
@@ -82,7 +82,7 @@ endif (CMAKE_BUILD_TYPE MATCHES Debug)
 install(
   TARGETS libopencc
-  LIBRARY DESTINATION ${DIR_LIBRARY}
-  ARCHIVE DESTINATION ${DIR_LIBRARY}
+  LIBRARY DESTINATION ${LIB_INSTALL_DIR}
+  ARCHIVE DESTINATION ${LIB_INSTALL_DIR}/static
 )
 
 install(
