--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -266,11 +267,6 @@ if (CMAKE_BUILD_TYPE MATCHES Debug)
   target_link_libraries(qt-version ${QT_LIBS})
 endif ()
 
-add_executable(web-demo tests/web_demo.cpp)
-target_link_libraries(web-demo
-                      ${LINK_LIBS}
-                      ${CEF_TARGET})
-
 ## Installation
 install(TARGETS "${CEF_TARGET}"
         DESTINATION "${QCEF_INSTALL_PATH}")
