diff -up x265_1.6/source/CMakeLists.txt.armhfp x265_1.6/source/CMakeLists.txt
--- x265_1.6/source/CMakeLists.txt.armhfp	2015-04-15 13:26:46.995287699 +0200
+++ x265_1.6/source/CMakeLists.txt	2015-04-15 13:27:29.015472793 +0200
@@ -51,9 +51,13 @@ if("${SYSPROC}" STREQUAL "" OR X86MATCH
         add_definitions(-DX86_64=1)
     endif()
 elseif(${SYSPROC} STREQUAL "armv6l")
-    message(STATUS "Detected ARM target processor")
+    message(STATUS "Detected ARMv6 target processor")
     set(ARM 1)
     add_definitions(-DX265_ARCH_ARM=1 -DHAVE_ARMV6=1)
+elseif(${SYSPROC} STREQUAL "armv7l")
+    message(STATUS "Detected ARMv7 system processor")
+    set(ARM 1)
+    add_definitions(-DX265_ARCH_ARM=1 -DHAVE_ARMV7=1)
 else()
     message(STATUS "CMAKE_SYSTEM_PROCESSOR value `${CMAKE_SYSTEM_PROCESSOR}` is unknown")
     message(STATUS "Please add this value near ${CMAKE_CURRENT_LIST_FILE}:${CMAKE_CURRENT_LIST_LINE}")
