Index: gst-plugins-ugly-0.10.19/common/gtk-doc-plugins.mak
===================================================================
--- gst-plugins-ugly-0.10.19.orig/common/gtk-doc-plugins.mak
+++ gst-plugins-ugly-0.10.19/common/gtk-doc-plugins.mak
@@ -224,9 +224,6 @@ html-build.stamp: sgml.stamp $(DOC_MAIN_
 	  fi; \
 	fi; \
 	cd html && gtkdoc-mkhtml $$mkhtml_options $(DOC_MODULE) $(DOC_MAIN_SGML_FILE)
-	@mv html/index.sgml html/index.sgml.bak
-	@$(SED) "s/ href=\"$(DOC_MODULE)\// href=\"$(DOC_MODULE)-@GST_MAJORMINOR@\//g" html/index.sgml.bak >html/index.sgml
-	@rm -f html/index.sgml.bak
 	@rm -f html/$(DOC_MAIN_SGML_FILE)
 	@rm -rf html/xml
 	@rm -f html/version.entities
Index: gst-plugins-ugly-0.10.19/common/gtk-doc.mak
===================================================================
--- gst-plugins-ugly-0.10.19.orig/common/gtk-doc.mak
+++ gst-plugins-ugly-0.10.19/common/gtk-doc.mak
@@ -131,9 +131,6 @@ html-build.stamp: sgml.stamp $(DOC_MAIN_
 	  mkhtml_options=--path="$(abs_srcdir)"; \
 	fi; \
 	cd html && gtkdoc-mkhtml $$mkhtml_options $(MKHTML_OPTIONS) $(DOC_MODULE) ../$(DOC_MAIN_SGML_FILE)
-	@mv html/index.sgml html/index.sgml.bak
-	@$(SED) "s/ href=\"$(DOC_MODULE)\// href=\"$(DOC_MODULE)-@GST_MAJORMINOR@\//g" html/index.sgml.bak >html/index.sgml
-	@rm -f html/index.sgml.bak
 	@rm -rf html/xml
 	@rm -f version.entities
 	@test "x$(HTML_IMAGES)" = "x" ||  ( cd $(srcdir) && cp $(HTML_IMAGES) $(abs_builddir)/html )
Index: gst-plugins-ugly-0.10.19/docs/plugins/Makefile.in
===================================================================
--- gst-plugins-ugly-0.10.19.orig/docs/plugins/Makefile.in
+++ gst-plugins-ugly-0.10.19/docs/plugins/Makefile.in
@@ -914,9 +914,6 @@ update:
 @ENABLE_GTK_DOC_TRUE@	  fi; \
 @ENABLE_GTK_DOC_TRUE@	fi; \
 @ENABLE_GTK_DOC_TRUE@	cd html && gtkdoc-mkhtml $$mkhtml_options $(DOC_MODULE) $(DOC_MAIN_SGML_FILE)
-@ENABLE_GTK_DOC_TRUE@	@mv html/index.sgml html/index.sgml.bak
-@ENABLE_GTK_DOC_TRUE@	@$(SED) "s/ href=\"$(DOC_MODULE)\// href=\"$(DOC_MODULE)-@GST_MAJORMINOR@\//g" html/index.sgml.bak >html/index.sgml
-@ENABLE_GTK_DOC_TRUE@	@rm -f html/index.sgml.bak
 @ENABLE_GTK_DOC_TRUE@	@rm -f html/$(DOC_MAIN_SGML_FILE)
 @ENABLE_GTK_DOC_TRUE@	@rm -rf html/xml
 @ENABLE_GTK_DOC_TRUE@	@rm -f html/version.entities
