--- a/src/list.coffee
+++ b/src/list.coffee
@@ -78,6 +78,9 @@
       if manifestPath = CSON.resolve(path.join(directoryPath, child, 'package'))
         try
           manifest = CSON.readFileSync(manifestPath)
+      delete manifest.readme
+      for item of manifest
+        delete manifest[item] if item.match /^\_/
       manifest ?= {}
       manifest.name = child
       if options.argv.themes
@@ -174,7 +177,7 @@
     options = @parseOptions(options.commandArgs)
 
     if options.argv.json
-      @listPackagesAsJson(options, callback)
+      @listPackagesAsJson(options)
     else if options.argv.installed
       @listInstalledPackages(options)
       callback()
