--- a/node_modules/symbols-view/lib/tag-generator.js
+++ b/node_modules/symbols-view/lib/tag-generator.js
@@ -72,7 +72,7 @@ export default class TagGenerator {
   generate() {
     let tags = {};
     const packageRoot = this.getPackageRoot();
-    const command = path.join(packageRoot, 'vendor', `ctags-${process.platform}`);
+    const command = 'ctags';
     const defaultCtagsFile = path.join(packageRoot, 'lib', 'ctags-config');
     const args = [`--options=${defaultCtagsFile}`, '--fields=+KS'];
