--- a/lib/main.coffee
+++ b/lib/main.coffee
@@ -51,7 +51,7 @@
     projectPath = atom.project.getPaths()[0]
     if atom.workspace.getActivePaneItem()
       false
-    else if path.basename(projectPath) is '.git'
+    else if projectPath && path.basename(projectPath) is '.git'
       # Only attach when the project path matches the path to open signifying
       # the .git folder was opened explicitly and not by using Atom as the Git
       # editor.
