--- a/package.json
+++ b/package.json
@@ -32,7 +32,6 @@
     "keytar": "^3.0",
     "mv": "2.0.0",
     "ncp": "~0.5.1",
-    "npm": "3.10.5",
     "open": "0.0.4",
     "plist": "git+https://github.com/nathansobo/node-plist.git",
     "q": "~0.9.7",
--- a/src/apm-cli.coffee
+++ b/src/apm-cli.coffee
@@ -95,7 +95,7 @@

 printVersions = (args, callback) ->
   apmVersion =  require('../package.json').version ? ''
-  npmVersion = require('npm/package.json').version ? ''
+  npmVersion = require('../../npm/package.json').version ? ''
   nodeVersion = process.versions.node ? ''

   getPythonVersion (pythonVersion) ->
--- a/src/clean.coffee
+++ b/src/clean.coffee
@@ -14,7 +14,7 @@
   @commandNames: ['clean']

   constructor: ->
-    @atomNpmPath = require.resolve('npm/bin/npm-cli')
+    @atomNpmPath = require.resolve('/bin/npm')

   getDependencies: (modulePath, allDependencies) ->
     try
--- a/src/config.coffee
+++ b/src/config.coffee
@@ -11,7 +11,7 @@
   constructor: ->
     atomDirectory = apm.getAtomDirectory()
     @atomNodeDirectory = path.join(atomDirectory, '.node-gyp')
-    @atomNpmPath = require.resolve('npm/bin/npm-cli')
+    @atomNpmPath = require.resolve('/bin/npm')

   parseOptions: (argv) ->
     options = yargs(argv).wrap(100)
--- a/src/dedupe.coffee
+++ b/src/dedupe.coffee
@@ -16,8 +16,8 @@
     @atomDirectory = config.getAtomDirectory()
     @atomPackagesDirectory = path.join(@atomDirectory, 'packages')
     @atomNodeDirectory = path.join(@atomDirectory, '.node-gyp')
-    @atomNpmPath = require.resolve('npm/bin/npm-cli')
-    @atomNodeGypPath = require.resolve('node-gyp/bin/node-gyp')
+    @atomNpmPath = require.resolve('/bin/npm')
+    @atomNodeGypPath = require.resolve('/bin/node-gyp')

   parseOptions: (argv) ->
     options = yargs(argv).wrap(100)
--- a/src/install.coffee
+++ b/src/install.coffee
@@ -25,8 +25,8 @@
     @atomDirectory = config.getAtomDirectory()
     @atomPackagesDirectory = path.join(@atomDirectory, 'packages')
     @atomNodeDirectory = path.join(@atomDirectory, '.node-gyp')
-    @atomNpmPath = require.resolve('npm/bin/npm-cli')
-    @atomNodeGypPath = require.resolve('node-gyp/bin/node-gyp')
+    @atomNpmPath = require.resolve('/bin/npm')
+    @atomNodeGypPath = require.resolve('/bin/node-gyp')

   parseOptions: (argv) ->
     options = yargs(argv).wrap(100)
@@ -103,6 +103,7 @@
     installArgs.push('--silent') if options.argv.silent
     installArgs.push('--quiet') if options.argv.quiet
     installArgs.push('--production') if options.argv.production
+    installArgs.push('--legacy-bundling')

     if vsArgs = @getVisualStudioFlags()
       installArgs.push(vsArgs)
--- a/src/publish.coffee
+++ b/src/publish.coffee
@@ -18,7 +18,7 @@

   constructor: ->
     @userConfigPath = config.getUserConfigPath()
-    @atomNpmPath = require.resolve('npm/bin/npm-cli')
+    @atomNpmPath = require.resolve('/bin/npm')

   parseOptions: (argv) ->
     options = yargs(argv).wrap(100)
--- a/src/rebuild.coffee
+++ b/src/rebuild.coffee
@@ -13,7 +13,7 @@

   constructor: ->
     @atomNodeDirectory = path.join(config.getAtomDirectory(), '.node-gyp')
-    @atomNpmPath = require.resolve('npm/bin/npm-cli')
+    @atomNpmPath = require.resolve('/bin/npm')

   parseOptions: (argv) ->
     options = yargs(argv).wrap(100)
