--- a/src/command.coffee
+++ b/src/command.coffee
@@ -1,4 +1,5 @@
 child_process = require 'child_process'
+fs = require './fs'
 path = require 'path'
 _ = require 'underscore-plus'
 semver = require 'semver'
@@ -94,6 +95,11 @@ class Command
       unless @electronVersion?
         throw new Error('Could not determine Electron version')
 
+      electronNodeDir = path.join('/usr/<lib>/electron/node')
+      atomNodeDevDir = path.join(config.getAtomDirectory(), '.node-gyp', '.node-gyp', @electronVersion)
+      if fs.existsSync(electronNodeDir) and not fs.existsSync(atomNodeDevDir)
+        fs.copySync(electronNodeDir, atomNodeDevDir)
+
       callback()
 
   getResourcePath: (callback) ->
