--- a/deploy/core/main.js
+++ b/deploy/core/main.js
@@ -2,7 +2,7 @@
 
 var app = require('app'),  // Module to control application life.
     BrowserWindow = require('browser-window'),  // Module to create native browser window.
-    ipc = require("ipc"),
+    ipc = require("electron").ipcMain,
     optimist = require('optimist');
 
 // Keep a global reference of the window object, if you don't, the window will
@@ -49,7 +49,7 @@
   });
 
   // and load the index.html of the app.
-  window.loadUrl('file://' + __dirname + '/LightTable.html?id=' + window.id);
+  window.loadURL('file://' + __dirname + '/LightTable.html?id=' + window.id);
 
   // Notify LT that the user requested to close the window/app
   window.on("close", function(evt) {
@@ -59,7 +59,9 @@
 
   // Emitted when the window is closed.
   window.on('closed', function() {
-    windows[window.id] = null;
+    try {
+        windows[window.id] = null;
+    } catch(e) {}
   });
 
   return window;
